### ISSUE TITLE
Running system tests stops development Puma server

### ISSUE THREAD

**Issue by renchap** [rails/rails#28122]
### Steps to reproduce

New Rails app from master, create a system test, launch `rails s` in a terminal and `rake test:system` in another one.

### Expected behavior

System tests run fine, and the `rails s` server is still running

### Actual behavior

When the Puma server from system tests starts, it signals the other Puma (started by `rails s`), which then tries to restart and fails because it detects the other instance :

```
» rails s
=> Booting Puma
=> Rails 5.1.0.alpha application starting in development on http://localhost:3000
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.7.1 (ruby 2.3.3-p222), codename: Snowy Sagebrush
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://localhost:3000
Use Ctrl-C to stop
Started GET "/" for ::1 at 2017-02-23 12:00:38 +0100
Processing by PagesController#home as HTML
  Rendering pages/home.html.erb within layouts/application
  Rendered pages/home.html.erb within layouts/application (1.3ms)
Completed 200 OK in 4220ms (Views: 4207.6ms | ActiveRecord: 0.0ms)

* Restarting...
=> Booting Puma
=> Rails 5.1.0.alpha application starting in development on http://localhost:3000
=> Run `rails server -h` for more startup options
A server is already running. Check /Users/renchap/dev/rails-test-app/tmp/pids/server.pid.
Exiting
```

### Analysis

I havent looked at the details, but I think it is because you use the same pid file (`tmp/pids/server.pid`) for the system tests Puma. Changing it to something else will probably solve this. I can submit a PR if you confirm this is the correct fix.

### System configuration

**Rails version**:
5.1.0 from master

**Ruby version**:
2.3.3p222

**Comment by renchap** [comment]
This is probably for @eileencodes (sorry, I moved my Capybara tests to the system tests and I am hitting a few issues :) ).

Great work on this BTW, a very nice addition to Rails!

**Comment by eileencodes** [comment]
Hrm ok I will take a look at this as well. Thanks for the report!

**Comment by eileencodes** [comment]
Hey I looked into this and it turns out it's a bug in Puma. The system tests don't use the Rails pidfile and don't set a pidfile at all - so adding another won't change anything. The problem is the restart plugin is overwriting all of the pids, rather than just the one that exists. So every time you start a new process with it's own pid in a Rails app it restarts any existing processes. 

I've opened a PR with Puma to fix it. Hopefully that gets merged and released 🔜 

https://github.com/puma/puma/pull/1226
